app/modules/enduser_setup.c

```diff
@@ -580,7 +581,8 @@ static int enduser_setup_http_handle_credentials(char *data, unsigned short data
     return 1;
   }
 
-  struct station_config *cnf = os_zalloc(sizeof(struct station_config));
+
+  struct station_config *cnf = luaM_malloc(lua_getstate(), sizeof(struct station_config));
```

Comment:
Note that this doesn't clear out memory.... Also, you need to use luaM_free to free up this block as well....